src/logger.ts

```diff
@@ -0,0 +1,61 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { JestConfig } from './jest-types';
+
+/**
+ * Logger file that enables logging things just once. Does this by traversing the array of previously recorded
+ * logs to see if the exact same message has already been logged
+ * @type {any[]}
+ */
+
+const logs: any[] = [];
+let loggingDisabled: boolean = true;
+let logsFlushed: boolean = false;
+
+export function enableLoggingIfNeeded(config: JestConfig) {
+  if (
+    config.globals &&
+    config.globals['ts-jest'] &&
+    config.globals['ts-jest'].debug
+  ) {
```

Comment:
using an environment variable might be a better option (`process.env.TS_JEST_DEBUG`, for example)

Reply:
Agree